### ISSUE TITLE
Compliance/Compliance Units Reduce an Obligation

### ISSUE THREAD

**Issue by NicoleGovvy** [bcgov/cas-compliance#70]
#### Description:

As an external user I can click Apply button on Apply Compliance Units page and the equivalent value of the units will decrease my obligation (fee adjustment in eLicensing).

[Apply Compliance Units page, Apply button](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=32714-8042&t=MnnBL99cdrS29s7t-4)

**Context:** compliance units (offsets or earned credits) moved from the BCCR holding account to compliance account will be considered to be used against the compliance obligation. 

The Apply button on the Apply Compliance Units page will trigger the BCCR end points for Account Creation (if OBPS Compliance sub-account does not exist in BCCR and bind the BOROID of the Operation) and transfer the units selected to this account in Active status. 

Upon success response from BCCR, the transaction will reduce the obligation balance in BCIERS and trigger the fee end point in eLicensing to decrease the fee with the equivalent amount. 

#### Acceptance Criteria:

Given that external user has selected CU
When they click apply button
Then CU will be transferred in their OBPS Compliance sub-account in BCCR and the operation obligation will decrease with the equivalent value

#### Development Checklist:

- [x] A new service (service/compliance_adjustment_service) is created
- [x] the `refresh_data_wrapper_by_compliance_report_version_id()` function can be made to "force_refresh" the data via a parameter (parameter should be default False)
- [x] In the new service, there is a create_adjustment function that calls `elicensing_api_client.adjust_fees` to create an adjustment
- [x] On successful creation of an adjustment in elicensing, the service force-refreshes the elicensing data using the refresh wrapper in `service/elicensing/elicensing_data_refresh_service`
- [x] Upon a successful call to BCCR to transfer the credits ( in `service/apply_compliance_units_service`) the new adjustment service is called to create the adjustment in elicensing
- [ ] Meets the DOD

#### Definition of Ready (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

#### Definition of Done (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Blocked By/Blocking
Blocked by #149 

#### Notes:

-

**Comment by patrickisaac** [comment]
Refinement Notes:
- May be split up into multiple tickets:
- 1. Transfer of info to eLicensing
- 2. cc: @vesselak / @vivanovbc

**Comment by fviduya** [comment]
I have tested this in the DEV environment, [here is the test document for reference](https://buttoninc-my.sharepoint.com/:w:/g/personal/francesca_button_is/EUiqNwlgi3hGok8_4Qodco4BFoesi7-h72sGLt2DlzLD3A?e=2VoCyU&nav=eyJoIjoiMTA4Njg3NjgxMCJ9).

I can verify that:
- [x] Given that external user has selected `Apply Compliance Units`, when they click the `Apply` button, the equivalent amount of the compliance units applied decreases the outstanding balance
- [x] When user clicks `Generate Compliance Invoice`, the equivalent value of the applied compliance units is visible in the generated PDF

Moving to Pending PO for review of test. Thank you!

**Comment by patriciarussellCAS** [comment]
Reviewed testing document. Moving to accepted.
